File: Livraison/Livraison/Livraison/Models/Delivery.cs
namespace Livraison.Models
{
    public class Delivery
    {
 
        public string Id { get; set; }
        public string IdComande { get; set; }
        public string IdClient { get; set; }
        public string IdLivreur { get; set; }
        public string Adresse { get; set; }
        public string Ville { get; set; }
        public string Status { get; set; }
        public string Prix { get; set; }
        public DateTime DateLivraison { get; set; }


    }

}

File: Livraison/Livraison-Api/models/Delivery.js
const mongoose = require('mongoose');

// Définition du schéma pour Delivery
const deliverySchema = new mongoose.Schema({
  id: { type: String, required: true },  // Id de la livraison
  idComande: { type: String, required: true },  // Id de la commande
  idClient: { type: String, required: true },  // Id du client
  idLivreur: { type: String, required: true },  // Id du livreur
  adresse: { type: String, required: true },  // Adresse de livraison
  ville: { type: String, required: true },  // Ville de livraison
  status: { type: String, required: true },  // Statut de la livraison
  prix: { type: String, required: true },  // Prix de la livraison
  dateLivraison: { type: Date, required: true }  // Date de livraison
});

// Index unique pour garantir qu'il n'y ait pas de doublons sur l'ID de la livraison
deliverySchema.index({ id: 1 }, { unique: true });

// Exportation du modèle Delivery
module.exports = mongoose.model('DELIVERY', deliverySchema);
